README.md
# AAAlertController

![logo](images/logo.png)

[![CI Status](http://img.shields.io/travis/aozhimin/AAAlertController.svg?style=flat)](https://travis-ci.org/aozhimin/AAAlertController)
[![Version](https://img.shields.io/cocoapods/v/AAAlertController.svg?style=flat)](http://cocoapods.org/pods/AAAlertController)
[![License](https://img.shields.io/cocoapods/l/AAAlertController.svg?style=flat)](http://cocoapods.org/pods/AAAlertController)
[![Platform](https://img.shields.io/cocoapods/p/AAAlertController.svg?style=flat)](http://cocoapods.org/pods/AAAlertController)

# Overview
`AAAlertController` is a lightweight UI component to present a popup style dialog, it provide alert style and action sheet style. It's easy to use and integrate in your project. The API is just like `UIAlertController` which you will see the usage in demo

## Sample project
We provided sample project, To run the it, after you clone the repo, open the workspace file.

![Demo](images/demo.gif)

## Requirements
`AAAlertController ` works on iOS 7.0+ and requires ARC to build.


## Installation

`AAAlertController` is available through [CocoaPods](http://cocoapods.org). To install
it, simply add the following line to your Podfile:

```ruby
pod "AAAlertController"
```

and edit pod file, install `AAAlertController` into your project just excute command as follow:

```ruby
pod install
```

## Usage
(see the usage which I put in the sample project)

### AAAlertStyleAlert

```objective-c
    AAPresentViewController *popupVC = [[UIStoryboard storyboardWithName:@"Main" bundle:nil]
                                        instantiateViewControllerWithIdentifier:@"AAAlertStyleViewController"];
        popupVC.view.frame = CGRectMake(0, 0, 300, 280);
    AAAlertController *alertController =
    [[AAAlertController alloc] initWithContentViewController:popupVC
                                           andPreferredStyle:AAAlertStyleAlert
                                          andAnimationOption:animationOption];
    [self presentViewController:alertController animated:YES completion:nil];
```

### AAAlertStyleStyleActionSheet

```objective-c
    AAPresentViewController *popupVC = [[UIStoryboard storyboardWithName:@"Main" bundle:nil]
                                        instantiateViewControllerWithIdentifier:@"AAActionSheetStyleViewController"];
        popupVC.view.frame = CGRectMake(0, 0, CGRectGetWidth(self.view.bounds), 200);
    AAAlertController *alertController =
    [[AAAlertController alloc] initWithContentViewController:popupVC
                                           andPreferredStyle: AAAlertStyleStyleActionSheet
                                          andAnimationOption:animationOption];
    [self presentViewController:alertController animated:YES completion:nil];
```

## Author
|Author|Gmail|Twitter|
|:---:|:----:|:----:|
|Alex Ao|aozhimin0811@gmail.com|[@Alex Ao](https://twitter.com/aozhimin0811)|

## License

`AAAlertController` is available under the MIT license. See the LICENSE file for more info.


CHANGELOG.md
## [0.2.0 - Action sheet style](https://github.com/aozhimin/AAAlertController/releases/tag/0.2.1) (2017-05-11)

#### Fix
* repalce UIDeviceOrientationDidChangeNotification to UIApplicationDidChangeStatusBarOrientationNotification

---

## [0.2.0 - Action sheet style](https://github.com/aozhimin/AAAlertController/releases/tag/0.2.0) (2017-04-18)

#### Add
* action sheet display style.

---

## [0.1.0 - AAAlertController, take off](https://github.com/aozhimin/AAAlertController/releases/tag/0.1.0) (2017-04-16)

First public release.


Example/AAAlertController/AAViewController.m
//
//  AAViewController.m
//  AAAlertController
//
//  Created by aozhimin on 04/14/2017.
//  Copyright (c) 2017 aozhimin. All rights reserved.
//

#import "AAViewController.h"
#import "AAAlertController.h"

typedef NS_ENUM(NSUInteger, AATableViewSectionType) {
    AATableViewSectionTypeAlert = 0,
    AATableViewSectionTypeActionSheet,
};

typedef NS_ENUM(NSUInteger, AATableViewCellType) {
    AATableViewCellTypeFade = 0,
    AATableViewCellTypeBounce,
};

static NSString *const AATableViewSectionTypeDescriptions [] = {
    [AATableViewSectionTypeAlert]       = @"Alert",
    [AATableViewSectionTypeActionSheet] = @"ActionSheet",
};

static NSString *const AATableViewCellTypeDescriptions [] = {
    [AATableViewCellTypeFade]   = @"Fade",
    [AATableViewCellTypeBounce] = @"Bounce",
};

@interface AAViewController () <UITableViewDataSource, UITableViewDelegate>

@property (nonatomic, strong) AAAlertController *alertController;

@end

@implementation AAViewController

- (void)viewDidLoad {
    [super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
}

#pragma mark - UITableView DataSource && Delegate

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return sizeof(AATableViewSectionTypeDescriptions) / sizeof(AATableViewSectionTypeDescriptions[0]);
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return sizeof(AATableViewCellTypeDescriptions) / sizeof(AATableViewCellTypeDescriptions[0]);
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    static NSString *identifier = @"tableViewCellIdentifier";
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:identifier
                                                            forIndexPath:indexPath];
    cell.textLabel.text = [NSString stringWithFormat:@"%@ Animation",
                           AATableViewCellTypeDescriptions[indexPath.row]];
    return cell;
}

- (NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInteger)section {
    return [NSString stringWithFormat:@"%@ Style", AATableViewSectionTypeDescriptions[section]];
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
    
    AAAlertStyle style;
    UIViewController *popupVC;
    if (indexPath.section == AATableViewSectionTypeAlert) {
        style = AAAlertStyleAlert;
        popupVC = [[UIStoryboard storyboardWithName:@"Main" bundle:nil]
                                            instantiateViewControllerWithIdentifier:@"AAAlertStyleViewController"];
        popupVC.view.frame = CGRectMake(0, 0, 300, 280);
    } else if (indexPath.section == AATableViewSectionTypeActionSheet) {
        style = AAAlertStyleStyleActionSheet;
        popupVC = [[UIStoryboard storyboardWithName:@"Main" bundle:nil]
                   instantiateViewControllerWithIdentifier:@"AAActionSheetStyleViewController"];
        popupVC.view.frame = CGRectMake(0, 0, CGRectGetWidth(self.view.bounds), 200);
    }
    
    AAAlertAnimationOptions animationOption;
    if (indexPath.row == AATableViewCellTypeFade) {
        animationOption = AAAlertAnimationOptionFade;
    } else if (indexPath.row == AATableViewCellTypeBounce) {
        animationOption = AAAlertAnimationOptionBounce;
    }
    
    AAAlertController *alertController =
    [[AAAlertController alloc] initWithContentViewController:popupVC
                                           andPreferredStyle:style
                                          andAnimationOption:animationOption];
    if (indexPath.section == AATableViewSectionTypeActionSheet) {
        alertController.backgroundDismiss = YES;
    }
    [self presentViewController:alertController animated:YES completion:nil];
}

@end
